x/farming/types/plan.go

```diff
@@ -124,6 +129,9 @@ func (plan BasePlan) Validate() error {
 	if _, err := sdk.AccAddressFromBech32(plan.TerminationAddress); err != nil {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid termination address %q: %v", plan.TerminationAddress, err)
 	}
+	if len(plan.Name) > MaxNameLength {
+		return sdkerrors.Wrapf(ErrInvalidNameLength, "plan name cannot be longer than max length of %d", MaxNameLength)
+	}
```

Comment:
I think we need to put in logic to check that the name trying to add does not duplicate the existing plans.
The number of plans will not be that large, so rather than using indexing for duplication check, we can check using `GetAllPlan()`

Reply:
I think adding validation for duplicate plan name requires an access to keeper. Are you suggesting to add the validation in `HandlePublicPlanProposal` in `keeper/proposal_handler.go`?

Reply:
I am thinking of adding these lines using `GetAllPlans()`. Any other approach you have in mind?
```go
plans := k.GetAllPlans(ctx)
for _, plan := range plans {
	if plan.(*types.BasePlan).Name == p.Name {
		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "plan name '%s' already exists", p.Name) 
	}
}
```